jetson/percep/test_protocol.md

```diff
@@ -0,0 +1,60 @@
+## Compile Test
+- [ ] ` ./jarvis build jetson/percep -o with_zed=false ar_detection=true obs_detection=true perception_debug=true `
+  - [ ] No compiler errors or warnings
+- [ ] ` ./jarvis build jetson/percep -o with_zed=false ar_detection=true obs_detection=true perception_debug=false `
+  - [ ] No compiler errors or warnings
+- [ ] ` ./jarvis build jetson/percep -o with_zed=true ar_detection=true obs_detection=true perception_debug=false `
+  - [ ] No compiler errors, only ZED warnings
+- [ ] ` ./jarvis build jetson/percep -o with_zed=true ar_detection=true obs_detection=true perception_debug=true `
+   - [ ] No compiler errors, only ZED warnings
+
+## Offline Debug Test
+- [ ] `./jarvis build jetson/percep -o with_zed=false ar_detection=true obs_detection=true perception_debug=true`
+- [ ] Run code against test1 image set
+- [ ] Run code against test2 image set
+- Check for:
+   - [ ]  Left and right path are being calculated
+   - [ ]  Calculated bearing is reasonable given viewer output
+   - [ ]  AR Tag ID is 2 in test1 and 6 in test2
+   - [ ]  2 AR Tags detected simultaneously test2
+   - [ ]  Obstacles with large width are detected
+   - [ ]  Path is projected around not through large obstacles
+   - [ ]  Short obstacles are detected
+   - [ ]  Steep inclines are detected as obstacles
+  
+## Offline Silent Test
+- [ ] `./jarvis build jetson/percep -o with_zed=false ar_detection=true obs_detection=true perception_debug=false`
+- [ ] Run `time ./jarvis exec jetson/percep` against test_case1 image set
+- [ ] Run `time ./jarvis exec jetson/percep` against test_case2 image set
+- [ ] User value is less than or equal to \<TODO Determine Value\>
+
+## Unit Test
+- [ ] ` ./jarvis build jetson/percep -o with_zed=true ar_detection=true obs_detection=true perception_debug=true `
+- [ ] `./jarvis exec lcm_bridge/server/`
+- [ ] `./jarvis exec lcm_tools_echo Obstacle "/obstacle"`
+- [ ] `./jarvis exec lcm_tools_echo TargetList "/target_list"`
+- Test environment contains:
+  - [ ] 1 large obstacle ~ over 40cm tall
+  - [ ] 2 small obstacles ~  25cm-30cm tall
+  - [ ] 1 post
+  - [ ] 1 hill
+- Test for:
+  - [ ] Detected large obstacle 6m away
+  - [ ] Detected small obstacle 2m away
+  - [ ] Detected clear path when 1.2m gap between 1 large and 1 small obstacle
+  - [ ] Left and right path are being calculated
+  - [ ] Calculated bearing is reasonable given viewer output of path
+  - [ ] AR Tag ID is the correct ID
+  - [ ] Obstacles with large width are detected
+  - [ ] Short obstacles are detected
+  - [ ] FPS values don't exceed \<TODO find this value\>
+  - [ ] LCM messages are populating properly
+
+## Integration Test
+- [ ] Create new branch based off of main
+- [ ] Cherry-Pick all PRs that need testing onto new branch
+- [ ] Checkout new branch
+- [ ] Compile all Software Packages
```

Comment:
Will folks know how to do this? Maybe just add the command that will do this. I can't remember it off the top of my head right now but there's a way to make Jarvis run everything.

Reply:
Our testing setup requires many packages to run at once. We could use the ./jarvis build -a tho that wouldn't run all of the required packages. Localization seems to have a lot of packages to run as well as has to ssh into the beaglebone to run stuff on there. We also have to run the gui server. Bottom line is I think there are a lot of commands required for launching the code and I'd rather someone make a document detailing the auton setup and linking that document, in our testing_protocol as opposed to adding all of the commands to our testing doc

Reply:
That all sounds good to me. Maybe just throw in the link to that other document? If it doesn't exist yet then just create an issue and add fixing this line to that issue (similar to the other comment in the PR referencing another issue).

Reply:
I don't think it exists, so I created Issue #554